src/Data/Vec/Recursive.agda

```diff
@@ -148,3 +155,18 @@ zip = zipWith _,_
 
 unzip : ∀ n → (A × B) ^ n → A ^ n × B ^ n
 unzip = unzipWith id
+
+private module ↔ {ℓ} = IsEquivalence (↔-isEquivalence {ℓ})
+
+lift↔ : {A B : Set a} → ∀ n → A ↔ B → A ^ n ↔ B ^ n
+lift↔ 0 AB = ↔.refl
+lift↔ 1 AB = AB
+lift↔ (2+ n) AB = ×-cong AB (lift↔ _ AB)
+
+Vec↔^ : ∀ n → Vec A n ↔ (A ^ n)
+Vec↔^ 0 = mk↔ ((λ _ → refl) , λ where Vec.[] → refl)
+Vec↔^ 1 = mk↔′ Vec.head Vec.[_] (λ _ → refl) (λ where (x ∷ Vec.[]) → refl)
+Vec↔^ {A = A} (2+ n) with Vec↔^ {A = A} (suc n)
+... | vec↔ =  mk↔′ (λ where (x ∷ xs) → x , to xs) (λ (x , xs) → x ∷ from xs)
```

Comment:
Do we need to `with`-abstract here? Can we write `where open Inverse (Vec↔^ {A = A} (suc n))`

Reply:
Yes. Without `with`-abstract, the termination checker does not work. If you know another solution to this problem, let me know.

Reply:
`let open Inverse (Vec↔^ (suc n)) in (...)` is probably enough.

Reply:
I removed this part because I saw that it is already in the library.